cmds/rm/rm_test.go

```diff
@@ -7,114 +7,246 @@
 package main
 
 import (
-	"fmt"
+	"bufio"
 	"io/ioutil"
 	"os"
-	"path/filepath"
+	"path"
 	"testing"
 )
 
-type makeit struct {
-	n string      // name
-	m os.FileMode // mode
-	s string      // for symlinks or content
+type file struct {
+	name   string
+	delete bool
 }
 
-var tests = []makeit{
-	{
-		n: "hi1.txt",
-		m: 0666,
-		s: "",
-	},
-	{
-		n: "hi2.txt",
-		m: 0777,
-		s: "",
-	},
-	{
-		n: "go.txt",
-		m: 0555,
-		s: "",
-	},
+type rmTestCase struct {
+	name  string
+	files []file
+	i     bool
+	r     bool
+	f     bool
+	err   func(error) bool
+	stdin *fakeStdin
 }
 
-// initial files and folders (for testing)
-func setup() (string, error) {
-	fmt.Println(":: Creating simulating data...")
-	d, err := ioutil.TempDir(os.TempDir(), "hi.dir")
-	if err != nil {
-		return "", err
-	}
-
-	tmpdir := filepath.Join(d, "hi.sub.dir")
-	if err := os.Mkdir(tmpdir, 0777); err != nil {
-		return "", err
+var (
+	no       = newFakeStdin("no")
+	fbody    = []byte("Go is cool!")
+	tmpFiles = []struct {
+		name  string
+		mode  os.FileMode
+		isdir bool
+	}{
+
+		{
+			name:  "hi",
+			mode:  0755,
+			isdir: true,
+		},
+		{
+			name: "hi/one.txt",
+			mode: 0666,
+		},
+		{
+			name: "hi/two.txt",
+			mode: 0777,
+		},
+		{
+			name: "go.txt",
+			mode: 0555,
+		},
 	}
+)
 
-	for _, t := range tests {
-		if err := ioutil.WriteFile(filepath.Join(d, t.n), []byte("Go is cool!"), t.m); err != nil {
-			return "", err
+func TestRemove(t *testing.T) {
+	var (
+		nilerr    = func(err error) bool { return err == nil }
+		testCases = []rmTestCase{
+			{
+				name: "no flags",
+				files: []file{
+					{"hi/one.txt", true},
+					{"hi/two.txt", true},
+					{"go.txt", true},
+				},
+				err:   nilerr,
+				stdin: no,
+			},
+			{
+				name: "-i",
+				files: []file{
+					{"hi/one.txt", true},
+					{"hi/two.txt", false},
+					{"go.txt", true},
+				},
+				i:     true,
+				err:   nilerr,
+				stdin: newFakeStdin("y", "no", "yes"),
+			},
+			{
+				name: "nonexistent with no flags",
+				files: []file{
+					{"hi/one.txt", true},
+					{"hi/two.doc", true}, // does not exist
+					{"go.txt", false},
+				},
+				err:   os.IsNotExist,
+				stdin: no,
+			},
+			{
+				name:  "directory with no flags",
+				files: []file{{"hi", false}},
+				err: func(err error) bool {
+					pe, ok := err.(*os.PathError)
+					if !ok {
+						return false
+					}
+
+					return pe.Err.Error() == "directory not empty"
```

Comment:
I believe you may be looking for

pe.Err == syscall.ENOTEMPTY

Reply:
👍